core/src/main/java/com/linecorp/armeria/client/retry/RetryException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * A {@link RuntimeException} that is raised when a {@link RetryingClient} finishes to retry due to
+ * {@code defaultMaxAttempts} or {@link Backoff#withMaxAttempts(int)} that has reached the limit.
+ */
+public final class RetryException extends RuntimeException {
```

Comment:
How about renaming to `RetryGiveUpException`? Any better names in mind? /cc @anuraaga ;-)